apps/src/client.rs

```diff
@@ -227,50 +251,57 @@ pub fn connect(
                 break 'read;
             }
 
-            let (len, from) = match socket.recv_from(&mut buf) {
-                Ok(v) => v,
-
-                Err(e) => {
-                    // There are no more UDP packets to read, so end the read
-                    // loop.
-                    if e.kind() == std::io::ErrorKind::WouldBlock {
-                        trace!("recv() would block");
-                        break 'read;
-                    }
-
-                    return Err(ClientError::Other(format!(
-                        "recv() failed: {:?}",
-                        e
-                    )));
-                },
-            };
+            for event in &events {
+                let token: usize = event.token().into();
+                let socket = &sockets[token];
+                let (len, from) = match socket.recv_from(&mut buf) {
+                    Ok(v) => v,
+
+                    Err(e) => {
+                        // There are no more UDP packets to read, so end the read
+                        // loop.
+                        if e.kind() == std::io::ErrorKind::WouldBlock {
+                            trace!("recv() would block");
+                            break 'read;
+                        }
+
+                        return Err(ClientError::Other(format!(
+                            "recv() failed: {:?}",
+                            e
+                        )));
+                    },
+                };
 
-            trace!("got {} bytes", len);
+                trace!("got {} bytes", len);
```

Comment:
from experience, it helps to be as descriptive in these lines as possible. So like above let's consider if there is any way for the log line to discriminate _exactly what_ received the bytes.

Reply:
Did the same as above.